src/wallet/api/wallet2_api.h

```diff
@@ -1144,6 +1149,545 @@ struct Wallet
+    * param: index_major - account index
+    * param: strict -
+    * return: [minor subaddress index: {amount, {blocks to unlock, time to unlock} }, ... ]
+    */
+    virtual std::map<std::uint32_t, std::pair<std::uint64_t, std::pair<std::uint64_t, std::uint64_t>>> unlockedBalancePerSubaddress(std::uint32_t index_major, bool strict) const = 0;
+    // QUESTION : Can anyone help with this comment? There are other ones I'm not satisfied with, so if you have a better suggestion I'll change them.
+    /**
+    * brief: isTransferUnlocked - check if transfer is unlocked
+    * param: unlock_time -
+    * param: block_height -
+    * return: true if transfer is unlocked, else false
+    */
+    virtual bool isTransferUnlocked(std::uint64_t unlock_time, std::uint64_t block_height) const = 0;
+    /**
+    * brief: updatePoolState -
+    * outparam: process_txs - [ [tx, tx_id, double_spend_seen], ... ]
+    * param: refreshed - (default: false)
+    * param: try_incremental - (default: false)
+    * note: sets status error on fail
+    * This public method is typically called to make sure that the wallet's pool state is up-to-date by
+    * clients like simplewallet and the RPC daemon. Before incremental update this was the same method
+    * that 'refresh' also used, but now it's more complicated because for the time being we support
+    * the "old" and the "new" way of updating the pool and because only the 'getblocks' call supports
+    * incremental update but we don't want any blocks here.
+    *
+    * simplewallet does NOT update the pool info during automatic refresh to avoid disturbing interactive
+    * messages and prompts. When it finally calls this method here "to catch up" so to say we can't use
+    * incremental update anymore, because with that we might miss some txs altogether.
+    */
+    virtual void updatePoolState(std::vector<std::tuple<cryptonote::transaction, std::string, bool>> &process_txs, bool refreshed = false, bool try_incremental = false) = 0;
+    /**
+    * brief: processPoolState -
+    * param: txs - [ [tx, tx_id, double_spend_seen], ... ]
+    * note: sets status error on fail
+    */
+    virtual void processPoolState(const std::vector<std::tuple<cryptonote::transaction, std::string, bool>> &txs) = 0;
+    // TODO / QUESTION : How to translate the following types to a standard type for the API?
+    // - tools::wallet2::transfer_details & tools::wallet2::transfer_container (vector of tools::wallet2::transfer_details)
+    // - tools::wallet2::payment_details
+    // - tools::wallet2::confirmed_transfer_details
+    // - tools::wallet2::pool_payment_details
+    // - tools::wallet2::signed_tx
+    // - cryptonote::address_parse_info
+// TODO : wallet2::transfer_container
+    /**
+    * brief: getTransfers - get all transfers
+    * outparam: transfers -
+    */
+//    virtual void getTransfers(wallet2::transfer_container& transfers) const = 0;
```

Comment:
Here we probably enter a whole region of things that we won't need in this form if my proposal to make a single encompassing "get payments / transactions" methods really flies.